app/controllers/solidus_paypal_commerce_platform/orders_controller.rb

```diff
@@ -1,14 +1,24 @@
 module SolidusPaypalCommercePlatform
   class OrdersController < ::Spree::Api::BaseController
     before_action :load_order
-    before_action :load_payment_method
+    before_action :load_payment_method, only: :create
     skip_before_action :authenticate_user
 
     def create
       authorize! :update, @order, order_token
       render json: @payment_method.gateway.create_order(@order, @payment_method.auto_capture), status: :ok
     end
 
+    def update_address
+      authorize! :update, @order, order_token
+      address = SolidusPaypalCommercePlatform::PaypalAddress.new(@order).update_address(params[:address])
+      if address.errors.any?
+        render json: address.errors.full_messages, status: :unprocessable_entity
```

Comment:
Not really important, but might be more _railsy_ if we made PaypalAddress focus around the duplicated address and be the one receiving the call to `#update`, quacking a bit more like an activerecord model.

```rb
paypal_address = SolidusPaypalCommercePlatform::PaypalAddress.new(@order)

if paypal_address.update(params[:address])
  render json: {}, status: :ok
else
  render json: paypal_address.errors.full_messages, status: :unprocessable_entity
end
```

Reply:
🦆 Thanks!